driver/src/price_finding/naive_solver.rs

```diff
@@ -151,18 +154,29 @@ impl PriceFinding for NaiveSolver {
                     }
                     None => continue,
                 }
+                let (x_u, x_du) = x
+                    .objective_value(
+                        prices[x.buy_token as usize],
+                        prices[y.buy_token as usize],
+                        exec_buy_amount[i],
+                        exec_sell_amount[i],
+                    )
+                    .unwrap();
+                let (y_u, y_du) = y
+                    .objective_value(
+                        prices[y.buy_token as usize],
+                        prices[x.buy_token as usize],
+                        exec_buy_amount[j],
+                        exec_sell_amount[j],
+                    )
+                    .unwrap();
+                let u = x_u + y_u;
+                let du = x_du + y_du;
+                if u < du {
```

Comment:
Can we add a unit test for that?

Reply:
I've moved this to a separate issue #285, for this PR we can just ignore the objective value calculation in the naive solver.